storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -1577,6 +1592,9 @@ write_binlog_info(MYSQL *connection)
 	gtid = (gtid_executed != NULL ? gtid_executed : gtid_current_pos);
 
 	if (mariadb_gtid || mysql_gtid) {
+		if (mysql_gtid && mysql_server_version >= 50705) {
+			flush_binlogs_maybe(connection);
```

Comment:
FLUSH BINARY LOGS potentially can take a while. It is done under either FTWRL or LOCK BINLOG FOR BACKUP which locks all commits, so it potentially can affect server performance. Have you measured this effect somehow?